File: leetcode/editor/en/[135]Candy.java
135
        Candy
        2023-01-07 10:25:02

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int candy(int[] ratings) {
        int n = ratings.length;
        int[] dp = new int[n];
        Arrays.fill(dp, 1);
        for (int i = 1; i < n; i++) {
            if (ratings[i] > ratings[i - 1]) {
                dp[i] = dp[i - 1] + 1;
            }
        }
        int sum = dp[n - 1];
        for (int i = n - 2; i >= 0; i--) {
            if (ratings[i] > ratings[i + 1]) {
                dp[i] = Math.max(dp[i], dp[i + 1] + 1);
            }
            sum += dp[i];
        }
        return sum;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: leetcode/editor/en/[135]Candy.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number[]} ratings
 * @return {number}
 */
var candy = function (ratings) {
    const n = ratings.length;
    let dp = new Array(n).fill(1);
    for (let i = 1; i < n; i++) {
        if (ratings[i] > ratings[i - 1]) {
            dp[i] = dp[i - 1] + 1;
        }
    }
    for (let i = n - 2; i >= 0; i--) {
        if (ratings[i] > ratings[i + 1]) {
            dp[i] = Math.max(dp[i], dp[i + 1] + 1);
        }
    }
    let sum = 0;
    for (const num of dp) {
        sum += num;
    }
    return sum;
};
//leetcode submit region end(Prohibit modification and deletion)
